#260: Rename 'Dataset extension' to 'Collection extension' + other improvements

m-mohr opened the pull request:
As discussed in #260.

- [x] Rough renaming in texts, samples and code
- [x] Rename the folder and files and all references
- [x] Read through all texts and make sure they still make sense
- [x] Re-check everything after the other parallel PRs are merged

This PR also fixes a bunch of issues in the examples found while reading through them.

hgs-truthe01 commented on extensions/README.md:
```diff
@@ -47,10 +46,9 @@ stable for over a year and are used in twenty or more implementations.
-| [EO](eo/) (`eo`)                                 | Item             | Covers data that represents a snapshot of the earth for a single date and time. It could consist of multiple spectral bands in any part of the electromagnetic spectrum. Examples of EO data include sensors with visible bands, infrared bands (near and shortwave), red edge bands and panchromatic bands. The extension provides com ...
-| [Scientific](scientific/) (`sci`)                            | Catalog +Dataset | Scientific metadata is considered to be data that indicate from which publication a dataset originates and how the dataset itself should be cited or referenced. | *Proposal* |
+| [EO](eo/) (`eo`)                                             | Item             | Covers electro-optical data that represents a snapshot of the earth for a single date and time. It could consist of multiple spectral bands, for example visible bands, infrared bands, red edge bands and panchromatic bands. The extension provides common fields like bands, cloud cover, off nadir, sun angle + elevati ...
+| [Scientific](scientific/) (`sci`)                            | Collection       | Scientific metadata is considered to be data that indicate from which publication a collection originates and how the collection itself should be cited or referenced. | *Proposal* |
```
I would agree with that.

m-mohr replied:
Changed with commit aab3f4073c342746ace37982b35ac5c41017cfec.

matthewhanson commented on catalog-spec/catalog-spec.md:
```diff
@@ -30,19 +31,17 @@ incorporated.
 There are two required element types of a Catalog: Catalog and Item. A STAC Catalog
 points to [STAC Items](../item-spec/), or to other STAC catalogs. It provides a simple
-linking structure that can be used recursively so that many `Items` can be included in 
+linking structure that can be used recursively so that many Items can be included in 
```
This isn't really true anymore. 
A Catalog is of course required, but if Items are required then Collections are required since Items must now point to a Collection. 
But we also talked about being able to create a Catalog containing a Collection that has no Items. So I think that Catalog and Collections are the two required elements, and Item is in fact optional.

m-mohr replied:
I think that's not true. Items don't require a Collection. Only Collections require Items to link back. But you could completely leave out collections and make a catalog just with only Catalogs and Items. See the discussion in https://github.com/radiantearth/stac-spec/pull/262#discussion_r223053198

It all depends on what you are using:
* Catalogs require Items.
* Collections don't require Items, but require Items to link back.
* Items don't require anything, not even a Catalog.

matthewhanson commented on collection-spec/README.md:
```diff
@@ -0,0 +1,64 @@
+STAC Collections are defined for flexibility. They only require a handful of fields, and
+implementors are free to add most any JSON field or object that they want via extensions.
+Many fields originating from the [STAC Item spec](../item-spec/item-spec.md) can already be 
+reused using the [Commons extension](../extentsions/commons/).
```
typo: extentsions

m-mohr replied:
Fixed.

matthewhanson commented on extensions/scientific/README.md:
```diff
@@ -1,10 +1,10 @@
-metadata is considered to be data that indicate from which publication a dataset originates and how
-the dataset itself should be cited or referenced. Overall, it helps to increase reproducibility and
+metadata is considered to be data that indicate from which publication a collection (dataset) originates and how
+the collection itself should be cited or referenced. Overall, it helps to increase reproducibility and
```
Update if sci is changed to Item scope.

m-mohr replied:
Updated with aab3f4073c342746ace37982b35ac5c41017cfec

matthewhanson commented on item-spec/item-spec.md:
```diff
@@ -111,12 +111,12 @@ whenever possible. But Static Catalogs are potentially more portable if they can
-provide additional information about a set of items, for example the license and provider information 
-and optionally any common information shared across all items, giving context on the overall set of
-data that an individual Item is a part of..
+Items are *strongly recommended* to provide a link to a STAC Collection definition. It is important as Collections 
```
Items are required now to link to a Collection

m-mohr replied:
That's not completely true, see my comment above ( https://github.com/radiantearth/stac-spec/pull/280#discussion_r224720185 ) - but I improved the wording here.

matthewhanson commented on item-spec/item-spec.md:
```diff
@@ -91,13 +91,13 @@ allowed to add additional fields such as a `title` and `type`.
+| self         | **REQUIRED.** _Absolute_ URL to the item file itself. This is required, to represent the location that the file can be found online. This is particularly useful when in a download package that includes metadata, so that the downstream user can know where the data has come from. |
+| root         | URL to the root STAC [Catalog](../catalog-spec/) or [Collection](../collection-spec/). |
+| parent       | URL to the parent STAC [Catalog](../catalog-spec/) or [Collection](../collection-spec/). |
+| collection   | STRONGLY RECOMMENDED. URL to a [Collection](../collection-spec/), which may use the [Commons extension](../extensions/commons/) and holds common fields of this and other Items (see chapter 'Collections'). |
```
Should be required if Items are required to link to a collection, or is only the link to be required?

m-mohr replied:
See my other comments for an explanation.

m-mohr commented:
* Changed the scope of the sci extension
* Further restructuring
* Improved/fixed several links
* Fixed issues raised here and improved wording regarding relationship between items/collections.

cholmes commented on collection-spec/collection-spec.md:
```diff
@@ -0,0 +1,113 @@
+
+**Note:** The [STAC Catalog specification](../catalog-spec/catalog-spec.md) requires a link to at least one `item` or `child` catalog. This is _not_ a requirement for collections, but _recommended_. In contrast to catalogs, it is **required** that items linked from a Collection MUST refer back to its Collection with the `collection` relation type.
+
+## Extensions
```
Why is the EO extension related? Because it uses bands? 

I don't think I love this section here, especially with no explanation as to why these related extensions are important. I'd prefer it either gone, or else have it actually discuss the relevance a bit, and make clear that the extensions are not mature yet - more ideas. Like mention their extension maturity here. If that's not there then it could feel like these are on the same level as the core spec.

m-mohr replied:
Removed EO and scientific, that were old references that more applied to the old dataset proposal. Added some word about the Commons extension, but could surely be improved further.

cholmes commented on extensions/scientific/README.md:
```diff
@@ -1,49 +1,51 @@
 [International DOI Foundation](https://www.doi.org/).
 
-- [Example](example-merraclim.json)
+- [Example (Collection)](example-merraclim.json)
```
Since this example depends on the common properties extension I'd mention that extension somewhere in the intro. And discuss if you have to do things that way, or if this also applies to items, etc. (it's ok to leave this out, since this is just at proposal stage, but I think it'd be good to have)

m-mohr replied:
Both are proposals, so I added some information to it. Scientific information are often more closely bound to the collection level anyway, so it should actually be recommended to use commons and collections to specify them there.

cholmes commented on api-spec/wfs-stac.md:
```diff
@@ -47,8 +48,9 @@ GET /collections
-spatial and temporal extents of all the data contained. A STAC search extension would only query those collections which
-have data that validates as STAC `Items` - with a datetime field and references to assets. But a STAC can live alongside
+spatial and temporal extents of all the data contained. The collections returned are compliant to both WFS Collections and 
+[STAC collections](../collections-spec/README.md). A STAC search extension would only query those collections which
```
I would ideally like a bit more discussion here, reiterating the point of the STAC collections, and contrasting them against WFS collections, perhaps mentioning any additional alignment we'd see in the future, etc. Not required to merge this by any means, but definitely a nice to have.

m-mohr replied:
Should be added at some point, but don't feel that I can really make a good proposal for this one now. Feel free to make a PR for it?!

m-mohr commented on item-spec/examples/sentinel2-sample.json:
```diff
@@ -7,7 +7,7 @@
     },
     {
-      "rel": "dataset",
+      "rel": "root",
```
Fixed.

m-mohr commented on item-spec/examples/sample-full.json:
```diff
@@ -33,7 +31,7 @@
     {"rel": "self", "href": "http://cool-sat.com/catalog/CS3-20160503_132130_04/CS3-20160503_132130_04.json"},
     {"rel": "thumbnail", "href":"thumbnail.png"},
-    {"rel": "dataset", "href": "http://cool-sat.com/catalog/"},
+    {"rel": "root", "href": "http://cool-sat.com/catalog/"},
```
Fixed.

m-mohr commented on item-spec/item-spec.md:
```diff
@@ -176,4 +177,37 @@ media type.
+that should be linked to in the Asset Object or in a Link Object, or even a new Asset Object could be added that is potentially easier to parse.
+There is a lot of metadata that is only of relevance to advanced processing algorithms, and while that is important it should not be in the core STAC items.
+
+### Asset definition
```
Added to #289.

m-mohr commented on item-spec/item-spec.md:
```diff
@@ -111,12 +111,13 @@ whenever possible. But Static Catalogs are potentially more portable if they can
-data that an individual Item is a part of..
+Items are *strongly recommended* to provide a link to a STAC Collection definition. It is important as Collections 
+provide additional information about a set of items, for example the license, provider information 
+and optionally any common information shared across all items using the [Commons extension](../extensions/commons/README.md),
```
Removed here, added to the extensions section with a note that it is only a proposal.

m-mohr commented:
I merged this PR now as I think I solved all important comments. Thanks for reviewing. If there's still something missing, I think it is easier to open a separate PR instead of using this rather large and messy PR.
